### ISSUE TITLE
Tab groups 

### ISSUE THREAD

**Issue by earlbellinger** [notepad-plus-plus/notepad-plus-plus#9517]
Chrome recently introduced [tab groups](https://blog.google/products/chrome/manage-tabs-with-google-chrome/). This would be a fantastic feature for Notepad++ to implement as well. #

**Comment by sasumner** [comment]
Notepad++ already has numerous ways to logically group tabs, e.g. "projects", "sessions", "folder as workspace", "favorites" (via plugin), and probably more... so I doubt such a thing as "tab groups" will ever be implemented.

**Comment by enricmm** [comment]
I concur with @earlbellinger, the tab groups as implemented in Chrome are a very easy way of on the fly grouping open tabs that can't be achieved (easily or conveniently) by the other options mentioned by @sasumner. 

Keep in mind Notepad++ is used by different breeds of users and on different things. While this might have initially been a "tool for developers" (mind my generalisation!) it has turned to be a very powerful text editor useful for a myriad of other purposes.

These days I don't develop anymore but I still use Notepad++ to keep track of many issues on a handful of different projects that I either coordinate or manage. I tend to stay away from project management tools when I can because none has the flexibility of Notepad++ to quickly add and, most importantly, to re-arrange information. For this reason, I might be handling over 50 different tabs, all open at the same time because I keep switching from one to the other on a daily basis. 

A feature like the one mentioned above would be ideal. it's a similar scenario to what browsing has become these days... dozens of tabs open at the same time which end up difficult to manage. So by allowing them into groups, Chrome has provided a great time-saving feature that also 'cleans-up' the mess provoked by so many open tabs.

In short, I'd love to see that on Notepad++, and if on top of that one could also colour the groups, then that'd be heaven!

**Comment by sasumner** [comment]
Some interesting ideas for "pinned tabs" and other tab-related things in #9699.
Not exactly the same as "groups" I guess, but somewhat related as possible future features.

**Comment by enricmm** [comment]
Thanks @sasumner

**Comment by klanderfri** [comment]
# Comment posted to indicate importance and interest of the request.
I agree with OP.
+1

**Comment by nielskool** [comment]
** Comment posted to indicate importance and interest of the request. **
I agree with OP.
+1

**Comment by ciltocruz** [comment]
+1

**Comment by korpx** [comment]
+1

**Comment by ayoub198fillali** [comment]
+1

**Comment by 0p71mu5** [comment]
+1

**Comment by hnavia** [comment]
New here.

I was looking for a feature like tabbed group but with a "small" twist. In my case, when working in different projects, I find myself using N++ as a regular notepad for taking notes or writing/adjusting information, codes and others. As I work in several projects at a time, I find myself creating new notes related to a project. The problem is that I might loose track of which new note belonged to which project.

I saw the "Directory as work area" (not sure the name in english) and I though it would be great to have a feature where I can create a Project and, under that project, have all the tabs that belongs to it. As some notes are not actually files (haven't saved them yet so they exist only as a temporal note) I will be able to quickly move it into this project with drag and drop.

I tried to find a solution or alternative but couldn't find any yet.

**Comment by 0p71mu5** [comment]
Grouping can also be implemented in Document List tab

**Comment by dhazell** [comment]
I also agree with this feature request. Tab groups would make it much easier to manage a high number of open text files

**Comment by thanh-guong** [comment]
+1

**Comment by thanh-guong** [comment]
+1

**Comment by vince-smdt** [comment]
+1, tab groups would be really useful when opening many files.

**Comment by AnDumu** [comment]
+1, or like workspaces like Opera, that when you switch between tabs are clear and open those were in that workspace

**Comment by tandreasen** [comment]
+1

**Comment by yb-cs** [comment]
+1

**Comment by Teasy** [comment]
the easiest (fastest) way to implement this seems to be:
- replicating the currently existing workspace/project panel (or similar)
- where all the currently opened tabs are visible (replicated from tab bar)
- then being able to make groups (like virtual folders in a workspace)
- and dragging existing tabs to any folder, and changing the order (like currently in workspaces)
- *without* affecting the actual tabs (and order) of the opened tabs (i.e. tab bar)
- this way, when new tabs are created, they just get added at the bottom of the "tabgroup" panel
- when clicking/opening a tab from the "tabgroup" panel, it simply switches to the existing tab
- close a tab from the "tabgroup" panel, simply invokes the "close tab" command
- this way, having the tab bar turned off, you can manage all tabs without even noticing the actual (internal) tab order
- you can even create a new set of keyboard shortcuts for: next/prev tab in tab group, and next/prev tab group
- this new "tabgroup" panel does not require to be saved, at all, like the current workspace/project panels are saved (see below)

this seems to be the closest to various other programs supporting tab groups, while modifying as little as possible of the existing notepad++ code, since this new tabgroup panel/functionality is mostly independent (apart from a few events like open/close tab, etc)

ideally, adding 1 or 2 new attributes in the session for each tab to store the location of the tab in the tabgroup
e.g. (simplified) <File filename="new 1" tabGroup="recipes" tabGroupIndex=0 />
this way, the tabgroup panel can be entirely populated from the saved session or workspace

**Comment by scofano** [comment]
+1

**Comment by DrRamaanand** [comment]
We are still looking forward to this feature. I hope someone develops it

**Comment by idibuseba** [comment]
+1

**Comment by noumanzakir** [comment]
+1

**Comment by anthonyleestark** [comment]
I've recently been thinking of this feature too.
It's good to know that a lot of people have requested it, I thought I had to make my own feature request and implement it myself =)))
Now I'm working on something else and don't really have time but I will try to work on it and update the result ASAP

**Comment by freezer2022** [comment]
> Chrome recently introduced [tab groups](https://blog.google/products/chrome/manage-tabs-with-google-chrome/).

Recently Firefox as well: [link](https://support.mozilla.org/en-US/kb/tab-groups).

**Comment by jrausch12** [comment]
+1

**Comment by anthonyleestark** [comment]
In case some of you have been waiting for any good news about this feature from me based on what I mentioned earlier, I’m sorry to disappoint you guys. The fact is, I can’t do anything about this feature anymore, or more specifically, I can no longer contribute to this project due to some personal reasons. 
Let’s hope someone else brings some good news soon.
